src/app/_shared-components/PostComments/Comments/Comments.tsx

```diff
@@ -42,11 +42,28 @@ function Comments({
 	const user = useAtomValue(userAtom);
 	const [showMore, setShowMore] = useState(false);
 	const [showSpam, setShowSpam] = useState(false);
+	const [showUnverified, setShowUnverified] = useState(false);
 	const [comments, setComments] = useState<ICommentResponse[]>(commentsFromProps);
+	const [identitiesLoaded, setIdentitiesLoaded] = useState<boolean>(false);
 
-	const regularComments = useMemo(() => comments.filter((comment) => !comment.isSpam), [comments]);
-	const spamComments = useMemo(() => comments.filter((comment) => comment.isSpam), [comments]);
-	const commentsToShow = showMore ? regularComments : regularComments.slice(0, 2);
+	interface CommentWithVerification extends ICommentResponse {
+		isVerified?: boolean;
+	}
+
+	const verifiedComments = useMemo(() => {
+		return comments.filter((comment) => !comment.isSpam && (comment as CommentWithVerification).isVerified);
+	}, [comments]);
+
+	const unverifiedComments = useMemo(() => {
```

Comment:
these three useMemo can be converted to one useMemo, return verified, unverified and spam comments from one useMemo, it will also minimise the array iterations